### ISSUE TITLE
Ranking Negativity of text

### ISSUE THREAD

**Issue by Stupack** [UKPLab/sentence-transformers#1312]
Hello, 

 I would like to teach a model **how negative a piece of text is**  (a score - the higher the more negative it is) given a set of text pairs. I suppose training a bi-encoder is the way to do it but Im not sure about the loss function. 

Pytorch provides the MarginRankingLoss function, and as far as I understand it seems similar to the ContrastiveLoss. Is there a better loss function for this use case? 

Thank you! :)

**Comment by nreimers** [comment]
Not sure I understand your task. You have a function f(text1, text2) that predicts the negativity of e.g. text1 given text2?

Is f(text1, text2) = f(text2, text1)? Or will this result in a different negativity value?

**Comment by Stupack** [comment]
I'd want to generate a score based on how negative a single text f(text1) is. 

I have a set of texts and I want to rank them (based on negativity). For each text, the only label I have is if the text is negative or not, I can create positive and negative texts pairs.
Training a cross-encoder using the pairings to tell me which text from each pair is more negative does not work because the model won't give me a score for single sentences.

So I guess the only option I have is to train a model in a siamese fashion where each side of the encoder receives a text and learns to rank each text.

Thanks for the help!

**Comment by nreimers** [comment]
CrossEncoder can also work for f(text1)
